### PR TITLE
Add types to ElkM1 library

### PR DESCRIPTION
**Pull request by gwww**
Added type information to the library (test, example, bin not typed yet).

This should not be a breaking change, however, I can imagine some uses of this library not using the types as defined.

Tempted to make this 2.0 of the lib. Thoughts?

The code has been run through both `mypy` and `pyright`. The config for `mypy` is in `mypy.ini`. The `mypy.ini` is taken from Home Assistant. Both come up with zero errors.

I'm unclear of the minimum Python version that will accept this typing. It is currently left unchanged at version 3.6

### CODE REVIEW

**Comment by gwww** [comment]
Hi @bdraco, would you mind taking a peek at the changes. I have completed a first pass on all files and `mypy` comes up clean. This is my first major kick at typing, so anything around best practices, etc would be helpful. Next step once this is done enough is to move onto the integration in HA.

**Comment by bdraco** [comment]
How about adding a `mypy` check to the CI?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 All notable changes to this project will be documented in this file.
 This project follows [Semantic Versioning](https://semver.org/).
 
+## [1.3.0]
+- Add typing
```
Comment (bdraco) [inline CHANGELOG.md:6]: Would be good to include a `py.typed` https://peps.python.org/pep-0561/

File: elkm1_lib/areas.py [hunk 2]
```diff
@@ -1,75 +1,83 @@
 """Definition of an ElkM1 Area"""
 
+from __future__ import annotations
```
Comment (bdraco) [inline elkm1_lib/areas.py:3]: Should change the minimum python version with this change

File: elkm1_lib/areas.py [hunk 3]
```diff
@@ -83,16 +91,16 @@ def _as_handler(self, armed_statuses, arm_up_states, alarm_states):
             self.elk.send(az_encode())
 
     def _ee_handler(
-        self, area, is_exit, timer1, timer2, armed_status
-    ):  # pylint: disable=too-many-arguments
-        area = self.elements[area]
+        self, msg_area: int, is_exit: bool, timer1: int, timer2: int, armed_status: str
+    ) -> None:
+        area: Area = cast(Area, self.elements[msg_area])
```
Comment (bdraco) [inline elkm1_lib/areas.py:96]:
```suggestion
        area = cast(Area, self.elements[msg_area])
```

The `: Area` is redundant with the cast

File: elkm1_lib/discovery.py [hunk 4]
```diff
@@ -87,19 +92,19 @@ class AIOELKDiscovery:
     BROADCAST_ADDRESS = "<broadcast>"
 
     def __init__(self) -> None:
-        self.found_devices: List[ElkSystem] = []
+        self.found_devices: list[ElkSystem] = []
```
Comment (bdraco) [inline elkm1_lib/discovery.py:95]: Be sure to import `from __future__ import annotations` with the new style typing

**Pull request merged**